src/Libraries/CoreNodeModels/Properties/Resources.en-US.resx

```diff
@@ -584,4 +584,22 @@
   <data name="SelectionPointsOutputPortName" xml:space="preserve">
     <value>Points</value>
   </data>
+  <data name="ListGroupByKeyDescription" xml:space="preserve">
+    <value>Group items into sub-lists based on their like key values</value>
+  </data>
+  <data name="ListGroupByKeyGroupsTooltip" xml:space="preserve">
+    <value>list of sublists, with items grouped by like key values</value>
+  </data>
+  <data name="ListGroupByKeyKeysTooltip" xml:space="preserve">
+    <value>Key values, one per item in the input list, used for grouping the items</value>
```

Comment:
yes, doubles have precision issues: comparing floating point numbers past a certain digit is not guaranteed to work, the same is true for other types of objects for other reasons, like you said similarity vs the same reference, I think 

Ienumerable.GroupBy, https://msdn.microsoft.com/en-us/library/vstudio/bb534501%28v=vs.100%29.aspx?f=255&MSPPError=-2147217396

uses the default equality comparer that the object itself defines, and then probably falls back to reference equality... I think this probably won't be too much of an issue unless a user is defining their own types to use as keys or using zero touch or python objects as keys that don't implement Equality  / getHashCode.

I've seen some posts on the forum where users are confused why extracting doubles like xyz locations don't work all the time for grouping.